tests/integration/test_extract.py

```diff
@@ -15,99 +15,102 @@
 
 import numpy as np
 import pytest
+from utils.generators import mk_seq_array
 
 import cunumeric as num
 
-np.random.seed(42)
-
-
-def test_extract():
-    cnp = np.array(
-        [1, 54, 4, 4, 0, 45, 5, 58, 0, 9, 0, 4, 0, 0, 0, 5, 0, 1]
-    ).reshape(
-        (6, 3)
-    )  # noqa E501
-    c = num.array(cnp)
-    bnp = np.random.randn(6, 3)
-    b = num.array(bnp)
-    assert num.array_equal(num.extract(c, b), np.extract(cnp, bnp))
-
-
-ARR = [
-    [1, 54, 4, 4, 0, 45, 5, 58, 0, 9, 0, 4, 0, 0, 0, 5, 0, 1],
-    [[1, 54, 4], [4, 0, 45], [5, 58, 0], [9, 0, 4], [0, 0, 0], [5, 0, 1]],
-    [
-        [[1, 54, 4], [4, 0, 45]],
-        [[5, 58, 0], [9, 0, 4]],
-        [[0, 0, 0], [5, 0, 1]],
-    ],
-    [[[1 + 2j, 54, 4], [4, 0 + 1j, 45]], [[5, 58, 0], [9, 0, 4]]],
-    [[True, False], [True, True], [True, False]],
-    [[]],
-    [],
-    [
-        [[0, 0, 0], [0, 0, 0]],
-        [[0, 0, 0], [0, 0, 1]],
-    ],
-    [False, False, False],
-    [
-        [[0, 0, 0], [0, 0, 0]],
-        [[0, 0, 0], [0, 0, 0]],
-    ],
```

Comment:
There are a lot of cases here that seem like they were explicitly chose for some reason or other, that don't appear to be part of the updated tests. Are we sure that the new tests exercise everything that the old tests did?

Reply:
After doing some tests, I screened some array in ARR and left them. At the same time, I found a new complex condition bug in the new submission.